### PR TITLE
[feat]: [PL-21667]: Adding pagination support for Accounts fetch and added unit test cases.

### PR DESCRIPTION
**Pull request by boopesh007**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)

### CODE REVIEW

**Review by shashank-harness — APPROVED** [review]
> Reviewed 2 of 5 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by pbarapatre10 — DISMISSED** [review]
> Reviewed 2 of 5 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: :shipit: complete! ✓

**Review by shashank-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: Some of the checks are too old. 2 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -427,10 +427,21 @@ public User createNewOAuthUser(User user, String accountId) {
...
+      if (user.getDefaultAccountId().equals(account.getUuid())) {
+        defaultAccount = account;
+        break;
+      }
+    }
+    if (defaultAccount != null) {
+      userAccounts.remove(defaultAccount);
+      userAccounts.add(0, defaultAccount);
+    }
     userAccounts.addAll(user.getSupportAccounts());
```
Comment (avinashmadhwani02) [inline 400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java:445]: Can default account be part of support accounts?

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java` line 445 at r4](https://reviewable.io/reviews/harness/harness-core/35097#-N6MCR_vJ4UVHfBZ-gWY-r4-445:-N7Q9wvz0UjOUlEnf4RK:b32w12) ([raw file](https://github.com/harness/harness-core/blob/d69ae390e6836c539adf3505ac0289993ecc1007/400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java#L445)):*
<details><summary><i>Previously, avinashmadhwani02 (Avinash Madhwani) wrote…</i></summary><blockquote>

Can default account be part of support accounts?
</blockquote></details>

No

**Review by shashank-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 5 files at r1, 1 of 3 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: 1. unresolved discussions.

**Review by shashank-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: 1. unresolved discussions.

**Review by shashank-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: 1. unresolved discussions.

**Review by avinashmadhwani02 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35097)* status: :shipit: complete! ✓

**Pull request merged**